ddtrace/profiling/profiler.py

```diff
@@ -301,28 +287,32 @@ def start(self):
         if self._scheduler is not None:
             self._scheduler.start()
 
-        self.status = ProfilerStatus.RUNNING
-
     def stop(self, flush=True):
         """Stop the profiler.
 
-        :param flush: Wait for the flush of the remaining events before stopping.
+        :param flush: Flush a last profile.
         """
+        if self.status != _service.ServiceStatus.RUNNING:
+            return
+
         if self._scheduler:
             self._scheduler.stop()
+            # Wait for the export to be over: export might need collectors (e.g., for snapshot) so we can't stop
+            # collectors before the possibly running flush is finished.
+            self._scheduler.join()
```

Comment:
we didn't have a timeout on the join before, but should we have one?

not blocking on this PR since is something that was preexisting.

Reply:
We *could* have one though the current code does not handle it.
Though there's no case where the scheduler join() would hang forever (or it's a bug we need to fix).